### PR TITLE
revamp CLI interface

### PR DESCRIPTION
**Pull request by rogpeppe**
Change the CLI interface to make it suitable for running go generate.
By default we generate just a single file containing the API
interface and the parameters that it uses.

When running a server, by default we generate the code to a temporary
directory and remove it afterwards.

We change the --port option to --http (mirroring the godoc flag)
so that it's possible to choose an address to listen on.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -28,30 +30,61 @@ var printCmdUsage = func() {
 	gnuflag.PrintDefaults()
 }
 
+var (
+	outputDir      = gnuflag.String("outputdir", "", "The output directory to save generated server package (default: the current directory, or a temporary directory if --server is specified")
+	serve          = gnuflag.Bool("serve", false, "If set, the generated server will be run after generation; implies --server")
+	listenAddr     = gnuflag.String("http", "", "Implies --server. If set, the generated server will be run after generation on the given network address")
```
Comment (mhemmings) [inline main.go]: `Implies --serve`

**Review by mhemmings — CHANGES_REQUESTED** [review]
I'm not sure the args are working as expected.
For example, running `openapi-httprequest --outputdir=output --serve spec.yaml` does not start the example server.
Furthermore, maybe rename `serve` and `server` to be more explicit?

**Comment by rogpeppe** [comment]
I've removed the `--serve` flag and fixed a couple of other things. PTAL.

File: go.mod [hunk 2]
```diff
@@ -18,7 +18,7 @@ require (
 	github.com/stretchr/testify v1.2.2 // indirect
 	golang.org/x/crypto v0.0.0-20181127143415-eb0de9b17e85 // indirect
 	golang.org/x/net v0.0.0-20181129055619-fae4c4e3ad76 // indirect
-	gopkg.in/errgo.v1 v1.0.0 // indirect
+	gopkg.in/errgo.v1 v1.0.0
```
Comment (mhemmings) [inline go.mod:21]: Why does my `go mod tidy` add this back in?

**Review by mhemmings — APPROVED** [review]
UX much nicer with these changes, thanks.

**Pull request merged**